### ISSUE TITLE
Incomplete removal of template VM

### ISSUE THREAD

**Issue by GWeck** [QubesOS/qubes-issues#5948]
**Qubes OS version**
R4.0.3

**Affected component(s) or functionality**
`qubes-dom0-update`

**Brief summary**
If a template VM was installed via the command `qubes-dom0-update`, its property `installed-by-rpm` is set to `true`. In this case, the Qube Manager refuses to delete this VM. It is, however, possible to clear the flag via the command
`qvm-prefs` _template-name_ `installed-by-rpm false`
and then remove the template VM via the Qubes Manager or the command `qvm-remove` _template-name_. In this case, the template and its menu entries are deleted, but the Qube Manager still lists the template in the `About -> Qubes OS -> Version information` display. If the template should be installed again via `qubes-dom0-update`, installation is aborted.

**To Reproduce**
1. Install a template via the command `qubes-dom0-update qubes-template-`_template-name_
2. Change its property via `qvm-prefs` _template-name_ `installed-by-rpm false`
3. Remove the template via the Qube Manager command `Qube -> Delete qube`
4. List the available system components via the Qube Manager command `About -> Qubes OS -> Version information`
5. Try to reinstall the template VM via the command `qubes-dom0-update qubes-template-`_template-name_

**Expected behavior**
1. The deleted VM should disappear from the list of Qubes components.
2. Reinstallation should succeed.

**Actual behavior**
1. The template is still displayed in the list.
2. Installation is aborted with the error message
`All matches were filtered out by exclude filtering for argument qubes-template-`_template-name_
`Errror: Unable to find a match: qubes-template-`_template-name_

**Additional context**
Reinstallation is unsuccessfull, too, if it is tried via `sudo dnf install`. Options like `--allowerasing`, `--skip-broken`, or `--best` have no effect.

I see three ways to mend this behaviour:
1. `qvm-remove` and the corresponding function `Qube --> Delete qube` of the Qube Manager might check for the entry in the component list and remove it, even if the flag `installed-by-rpm` is not set.
2. The error message on reinstallation of the template VM might be changed to a less misleading text, possibly stating that the template still seems to be present. This, however, may be an upstream issue, if the message is produced by `dnf`, coming from Fedora.
3. Anyway, it would be helpful, if this situation would be described in the template management part of the documentaion.

**Solutions you've tried**
The situation can be cleared by reinstalling via the command
`qubes-dom0-update --action=reinstall qubes-template-`_template-name_
and the removing the template VM via the command
`sudo dnf remove qubes-template-`_template-name_

**Relevant [documentation](https://www.qubes-os.org/doc/) you've consulted**
[https://www.qubes-os.org/doc/templates/#updating](url) and corresponding documentation

**Related, [non-duplicate](https://www.qubes-os.org/doc/reporting-bugs/#new-issues-should-not-be-duplicates-of-existing-issues) issues**
#3299, #4099, #4410, #4792

**Comment by GWeck** [comment]
I tested the above with Fedora templates. Debian seems to behave differently, as the `--action=reinstall` command, when trying to repair, fails with the error
`Installed package ... not available`
`Error: No packages marked for reinstall`
I will continue testing.

**Comment by GWeck** [comment]
For Debian, the offending entry seems to be deleted via the command
`rpm -e --noscripts qubes-template-`_template-name_
(after producing a lot of error messages about missing files in `/var/lib/qubes/vm-templates/`_template-name_)

**Comment by marmarta** [comment]
I think this will be (at least partially) helped by this PR: https://github.com/QubesOS/qubes-core-admin-client/pull/145 ; once it is ready, I'll integrate it into Qube Manager and other GUI tools.

**Comment by GWeck** [comment]
The hack for debian also works for centos.

**Comment by icequbes1** [comment]
I've also run into this after I ran out of disk space while installing the template. It's essentially the same problem - uninstallation of the rpm is not possible unless forcing `--noscripts`.

In my case the issue was reproduced as such:

1. Executed: `sudo qubes-dom0-update qubes-template-fedora-32`
2. During the postinstall of the package, `qvm-template-postprocess` errors out because I ran out of disk space in the LVM pool.
3. This leaves the system (rpm/dnf) believing qubes-template-fedora-32 is still installed when in fact it's in a half-installed state (for me, the root image part files was left at /var/lib/qubes/vm-templates/tmp-fedora-32.
4. Attempting to run `qubes-dom0-update qubes-template-fedora-32` always results in "No match for argument: qubes-template-fedora-32". This is because `--exclude=[list of installed templates as returned by rpm -qa]` is added to the `dnf install` command executed in the updatevm.
5. Removing the half-installed qubes-template-fedora-32 with `dnf` or `rpm` also fail during postuninstall - because qvm-template-postprocess returns "No Qube with this name exists". This prevents the rpm package from being uninstalled.
6. The only option is to attempt `dnf install/reinstall <template_package>` in hopes that it succeeds, or as a last resort, `rpm -e --noscripts <template_package>` (as already documented by @GWeck). This leaves the root fs images at /var/lib/qubes/vm-templates/tmp-[templatename], that the user must delete manually and is especially important if disk space issues caused the initial failure.
